content/blog/2024-07-31-valkey-8-0-0-rc1.md

```diff
@@ -0,0 +1,157 @@
+
+NOTE: Not all improvements are available in the release candidate, but they will
+be available in the GA release of Valkey 8.0.
+
+- **Asynchronous I/O Threading**: Enables parallel processing of commands and
+  I/O operations, maximizing throughput and minimizing bottlenecks.
+- **Intelligent Core Utilization**: Distributes I/O tasks across multiple
+  cores based on realtime usage, reducing idle time and improving energy efficiency.
+- **Command Batching**: Optimizes memory access patterns by prefetching frequently
+  accessed data to minimize CPU cache misses, reducing memory accesses required for
+  dictionary operations.
+
+For more details on these improvements, you can refer to
+[#758](https://github.com/valkey-io/valkey/pull/758) and
+[#763](https://github.com/valkey-io/valkey/pull/763).
+
+## Reliability
+
+Cluster scaling operations via slot migrations have historically been delicate.
+Valkey 8.0 improves reliability and minimizes disruptions with the following
+enhancements:
+
+- **Automatic Failover for Empty Shards**: New shards that start empty, owning
+  no slots, now benefit from automatic failover. This ensures high availability
+  from the start of the scaling process.
+- **Replication of Slot Migration States**: All `CLUSTER SETSLOT` commands are
+  now replicated synchronously to replicas before execution on the primary. This
+  reduces the chance of unavailability if the primary fails, as the replicas have
+  the most up-to-date information about the state of the shard. New Replicas also
+  automatically inherit the state from the primary without additional input from
+  an operator.
+- **Slot Migration State Recovery**: In the event of a failover, Valkey 8.0 automatically
+  updates the slot migration states on source and target nodes. This ensures requests
+  are continuously routed to the correct primary in the target shard, maintaining
+  cluster integrity and availability.
+
+For more details on these improvements, you can refer to
+[#445](https://github.com/valkey-io/valkey/pull/445).
+
+## Replication
+
+Valkey 8.0 introduces a dual-channel replication scheme, allowing the RDB and
+the replica backlog to be transferred simultaneously, accelerating synchronization.
+
+- **Reduced Memory Load**: By streaming replication data to the replica during
+  the full sync, the primary node experiences significantly less memory pressure.
+  The replica now manages the Client Output Buffer (COB) tracking, reducing the
+  likelihood of COB overruns and enabling larger COB sizes on the replica side.
+- **Reduced Parent Process Load**: A dedicated connection for RDB transfer frees
+  the primary's parent process from handling this data, allowing it to focus on
+  client queries and improving overall responsiveness.
+
+Performance tests show improvements in write latency during sync, and in scenarios
+with heavy read commands, the sync time can be cut by up to 50%. This translates
```

Comment:
do we have source for this? "up to" doesn't make me as worried, but just figured I'd note it.

Reply:
It's detailed in the issue linked below.